### PR TITLE
AnneJu step 6: Create Task model

### PR DESCRIPTION
**Pull request by anneju**
- [x] create task model
- [x] add gem `annotate` for better look of columns

### CODE REVIEW

**Comment by anneju** [comment]
As I am not familiar with views, I used Scaffold to create views for me. So there are many views created along in this PR 🙇🏻‍♀️

File: myapp/app/models/task.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# == Schema Information
+#
+# Table name: tasks
+#
+#  id          :bigint           not null, primary key
```
Comment (mt-ytaher) [inline myapp/app/models/task.rb:5]: we should use unsigned as we don't want negative value in `id`

i.e: `t.bigint :id, unsigned: true, null: false`

File: myapp/app/models/task.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#  title       :string(255)
+#  description :string(255)
+#  priority    :integer
+#  status      :integer
```
Comment (mt-ytaher) [inline myapp/app/models/task.rb:9]: for priority and status, you probably will have few fixed values and for that maybe `tinyint` is enough.

we can do something like `t.integer :priority, limit: 1`


For reference

```

:limit	    Numeric Type	Column Size	Max value
1	     tinyint	         1 byte	          127
2	     smallint	         2 bytes	32767
3	     mediumint	         3 byte	        8388607
nil, 4, 11    int(11)	         4 byte	       2147483647
5..8	      bigint	         8 byte	     9223372036854775807
```

File: myapp/db/migrate/20221212045433_create_tasks.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreateTasks < ActiveRecord::Migration[6.0]
+  def change
+    create_table :tasks do |t|
+      t.string :title
+      t.string :description
+      t.integer :priority
+      t.integer :status
```
Comment (mt-ytaher) [inline myapp/db/migrate/20221212045433_create_tasks.rb]: for priority and status, you probably will have few fixed values and for that maybe `tinyint` is enough.

we can do something like `t.integer :priority, limit: 1`

File: myapp/db/schema.rb [hunk 4]
```diff
@@ -10,6 +10,16 @@
...
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 0) do
+ActiveRecord::Schema.define(version: 2022_12_12_045433) do
+
+  create_table "tasks", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4", force: :cascade do |t|
+    t.string "title"
+    t.string "description"
+    t.integer "priority"
+    t.integer "status"
```
Comment (mt-ytaher) [inline myapp/db/schema.rb]: same as https://github.com/Fablic/training/pull/1461/files#r1046622634

**Comment by mt-ytaher** [comment]
@anneju Hey, I have added few comments. Please have a look.

File: myapp/app/models/task.rb (hunk 2 above)
Comment (anneju) [reply myapp/app/models/task.rb:9]: didn't know about this either. Thanks for sharing~

File: myapp/app/models/task.rb (hunk 1 above)
Comment (anneju) [reply myapp/app/models/task.rb:5]: cool I didn't know about unsigned. Learned~

File: myapp/config/application.rb [hunk 5]
```diff
@@ -15,5 +15,8 @@ class Application < Rails::Application
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+    config.generators do |generator|
+      generator.orm :active_record, primary_key_type: :unsigned_integer
```
Comment (anneju) [inline myapp/config/application.rb:19]: referred [stackoverflow](https://stackoverflow.com/a/65955717) to set all table id unsigned integer

File: myapp/app/models/task.rb (hunk 1 above)
Comment (anneju) [reply myapp/app/models/task.rb:5]: these notes are autogenerated by annotate gem, though it's not reflected we have changed the `int types` in these notes, it's been set [here](https://github.com/Fablic/training/pull/1461/files#r1046663783)

**Review by mt-ytaher — APPROVED** [review]
LGTM!!
[LGTM](https://lgtm.lol/i/671)

**Pull request merged**